specs/vep-2272-complete-data-job-configuration-persistence/README.md

```diff
@@ -95,75 +95,114 @@ There would be no changes to the public APIs.
+synchronized with the database, the process will initiate a new data job deployment. This deployment may involve tasks
+such as building an image or updating the Cron Job template.
+
+In order to enhance the process further, the proposed asynchronous process will incorporate a mechanism to determine
+whether a Cron Job needs to be updated or not. This will be achieved by utilizing the hash
+code (`deployment_version_sha`) of the
+Cron Job YAML configuration, which will be stored in the database after each deployment.
+
+By introducing this asynchronous process, you can ensure that the database and Kubernetes remain in sync, maintaining
+data integrity and enabling smooth operation of the system.
 
-### Kubernetes Cron Jobs Synchronizer Process
-
-In order to maintain data integrity and synchronization between database and Kubernetes we will introduce new
-asynchronous process. The process will be based on Spring Scheduler and Scheduler Lock similar
-to [DataJobMonitorCron.watchJobs()](https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMonitorCron.java#L84).
-
-Basically, it will iterate over the data job deployment configurations from the database and check whether
-the Cron Job is already synchronized or it needs to be synchronized. If it is not in sync with the database the process
-will initiate new data job deployment which may include building image or just changing the template. It will be
-based on the sha of the cronjob template, which will be generated right after the job deployment.
-
-<!--
-Dig deeper into each component. The section can be as long or as short as necessary.
-Consider at least the below topics but you do not need to cover those that are not applicable.
-
-### Capacity Estimation and Constraints
-    * Cost of data path: CPU cost per-IO, memory footprint, network footprint.
-    * Cost of control plane including cost of APIs, expected timeliness from layers above.
-### Availability.
-    * For example - is it tolerant to failures, What happens when the service stops working
-### Performance.
-    * Consider performance of data operations for different types of workloads.
-       Consider performance of control operations
-    * Consider performance under steady state as well under various pathological scenarios,
-       e.g., different failure cases, partitioning, recovery.
-    * Performance scalability along different dimensions,
-       e.g. #objects, network properties (latency, bandwidth), number of data jobs, processed/ingested data, etc.
 ### Database data model changes
-### Telemetry and monitoring changes (new metrics).
-### Configuration changes.
-### Upgrade / Downgrade Strategy (especially if it might be breaking change).
-  * Data migration plan (it needs to be automated or avoided - we should not require user manual actions.)
-### Troubleshooting
-  * What are possible failure modes.
```

Comment:
What are possibel failure modes . E.g what if DB is down, what if Control Service restart, what if single pod restart. 

Also who owns the syncing ? With multiple nodes I assume we will use sched lock for that but can you explain the logic. How would that work?

 A) it's better to be explicit and B) i do not recall it very well

Reply:
Added one sentence in the ### Kubernetes Cron Jobs Synchronizer and filled the ### Troubleshooting section.